hphp/runtime/vm/jit/vasm-util.cpp

```diff
@@ -29,7 +29,12 @@ bool is_nop(const copy& i) { return i.s == i.d; }
 bool is_nop(const copy2& i) { return i.s0 == i.d0 && i.s1 == i.d1; }
 
 // movb r,r is a nop, however movl is not since it zeros upper bits.
-bool is_nop(const movb& i) { return i.s == i.d; }
+bool is_nop(const movb& i) {
+  // On PPC64 movb r,r cannot be considered a trivial nop since it zeros
```

Comment:
@mxw Yes in fact I am doing some refactor here. I remove those `movb`instructions but what about `subbi` `testbi` `cmpbi` instructions I can't have garbage on the upper bits of register in this case as we are doing those operations with 64 bits. I don't know if i understand correctly but you say i can't `movb` to tmp registers and use it as Vreg64?